src/core/trust/mir/agent.cpp

```diff
@@ -241,10 +246,19 @@ core::trust::Request::Answer mir::Agent::authenticate_request_with_parameters(co
                     parameters.application.pid,
                     Agent::on_trust_session_changed_state,
                     &cb_context),
-        // Acquire a new fd for the prompt provider.
-        scope.prompt_session->new_fd_for_prompt_provider()
+        /* fd */ -1
     };
 
+    auto error = scope.prompt_session->error_message();
+    if (!error.empty()) {
+        throw std::runtime_error{
+            "Unable to create a prompt session: " + error
```

Comment:
Where should this message be logged?

Reply:
Perhaps here:

https://github.com/ubports/trust-store/blob/ad94b8bcdfeafc7ff7c80bc3693f8fecb4b9b70e/src/core/trust/dbus/agent.cpp#L75-L79

But I guess it can be another PR. There're throwing in a few other places too.